Add SOP0012 for Withdraw Dataset from Node Allele Frequency Beacon (#73)
GenomicDataInfrastructure/standard-operating-procedures: sops/node-specific/GDI-SOP0012_withdraw-dataset-from-node-allele-frequency-beacon.md

### 1. Document History

| Template Version | Instance version | Author(s)                   | Description of changes                                                                                                                                               | Date         |
| ---------------- | ---------------- | --------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------ |
| `v1`             |                  | Liina Nagirnaja, Oriol López-Doriga Sagalés | First version of the SOP for GitHub issue [#65](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/issues/65), based on the approved draft and reviewed copy. | `2026.04.10` |
| `v0`             |                  | Marcos Casado Barbero       | Created template for draft SOP                                                                 | `2026.03.05` |

### 2. Glossary

Find GDI SOPs common Glossary at the [**charter document**](../../docs/GDI-SOP_charter.md).

| Abbreviation  | Description                                       |

> | ------------- | ------------------------------------------------- |

Reviewer: In this document you are using **AF,** so maybe add this to glossary as well

| CRG           | Centre for Genomic Regulation                     |
| GDI           | European Genomic Data Infrastructure              |
| GDPR          | General Data Protection Regulation.               |
| EBI           | European Bioinformatics Institute                 |
| EMBL          | European Molecular Biology Laboratory             |
| FAIR          | Findability, Accessibility, Interoperability and Reusability             |
| HTTP          | Hypertext Transfer Protocol                       |

> | ID            | Identifier                                        |

Reviewer 2: I noticed by sheer chance a small discrepancy in this term with regards to the Charter, since the charter has ID as Identity Document.

Reviewer 2: But it's not an issue in your SOP, but rather an issue with SOP0008. What I would recommend is to change the Charter's definition, so that it's Identifier there as well. We'll change the SOP0008 afterwards.

| SOP           | Standard Operating Procedure                      |
| VHD           | Virtual Helpdesk                                  |

> | Term          | Definition                                                                                                |

Reviewer 2: The terms that overlap with the Charter should be the same as the charter if they are okay. For example, the ``Hard-deletion`` doesn't match the charter's definition. If you think the Charter's definition is wrong or incomplete (like in the case of the ``ID`` term above) you can propose a change to the charter glossary in this same PR.

Reviewer 2: once you've updated the Charter's definition of Query, for example, remember to update this one's too

Reviewer 2: I'm noticing the other terms don't match either, like ``Withdrawal``

| Soft-deletion | Data is marked as withdrawn and made inaccessible to users but retained internally for audit or limited-term retention.|
| Withdrawal    | A performance action to remove the dataset from beacon responses.                                         |

### 3. Roles and Responsibilities

See qualifications and responsibilities of the roles at the [**Organisational Roles and Responsibilities**](../../docs/GDI-SOP_organisational-roles-and-responsibilities.md) document.

| Role       | Full name                   | GDI/node role                                    | Organisation                          |
| ---------- | --------------------------- | ------------------------------------------------ | ------------------------------------- |
| Author     | Liina Nagirnaja             | Beacon Manager                                   | CRG                                   |
| Author     | Jordi Rambla                | Beacon Product Owner                             | CRG                                   |
| Author     | Oriol López-Doriga Sagalés  | Beacon Developer                                 | CRG                                   |

> | Reviewer   | Marcos Casado Barbero       | Task 4.3 member                                  | EMBL-EBI                              |

Reviewer 2: Credit where credit is due - @AlesCUMSI should be added as a reviewer too

Author: It reminded me of "Al César lo que es del César", it brought a smile to my face jeje.

| Approver   | Gabriele Rinck              | Task 4.3 member                                  | EMBL-EBI                              |
| Authorizer | Management Board            | Authorizer according to GDI SOP governance       | GDI                                   |

### 4. Purpose

AF Beacons expose population-level allele counts. A node-level SOP to withdraw data from Allele Frequency Beacons guarantees consistent, auditable removal of frequency contributions in line with GDPR and the GDI overarching withdrawal workflow.

### 5. Scope

This SOP covers the steps required to withdraw a dataset from a Node's Allele Frequency Beacon. Included steps are the identification of dataset records to withdraw, assessing the withdrawal scope, executing the removal of data, validating frequency integrity, auditing and logging the withdrawal, and reporting completion.

> It is part of the higher level Dataset Withdrawal SOP (SOP0009).

Reviewer 2: ```suggestion
It is part of the higher level Dataset Withdrawal SOP ([SOP0009](../european-level
/GDI-SOP0009_dataset-withdrawal.md)).
```

> ### 6. Introduction and Background Information

Reviewer 2: As a general comment on formatting, I would recommend to add extra newlines before each paragraph. Otherwise, markdown renders it all in one single paragraph and it's more difficult to read.

An aggregated data beacon hosts information for variants and minimal metadata for datasets that these variants belong to. These two types of different information are often stored separately in a database but linked through some reference property that contains the dataset id for the records. This dataset id is the pivotal piece of information, under which the records are displayed in a client response and that allows to integrate the response with other services in GDI.
When attempting a withdrawal of the beacon records related to a dataset, it will be necessary to determine which is the exact id of the dataset, as case sensitivity and other character inferences may result in an unsuccessful withdrawal attempt. The withdrawal procedure of a soft-deletion is slightly different from a hard-deletion, where the amount of records to be deleted is significantly higher. Withdrawal by hard-deletion includes variants and dataset records type, whereas soft-deletion includes only the latter. 

> It is important that the withdrawal procedure is preceded by a double-check of the dataset requested for withdrawal in coordination with the Fair Data Point service.

Reviewer 2: Suggestion to add a relative reference to the future SOP on FDP dataset withdrawal, which is the SOP0011. Even if not-existing for now.

Reviewer 2: Also, I'm unsure about this step. You are saying that there's a step to double check the dataset, and that this has to be also happening in the SOP0011, as in coordinated check? Or that the FDP withdrawal goes before this SOP0012?

Author: Yes, both. I think it's a coordinated check that needs to happen, as a beacon depends on the FDP to have the dataset so the dataset is shown in the Allele Frequency Browser. It would be weird to have a dataset in the FDP basket and then not see it in the Allele Frequency Browser, for me is an integrated update/withdrawal what needs to happen.

Reviewer 2: In theory the SOP0009 says these system-specific withdrawals go one by one, so I would expect they are coordinated. Nevertheless, never a bad thing to add a disclaimer saying "hey guys, remember to coordinate with whoever is executing the FDP withdrawal"

For a broader context of GDI SOPs, please refer to the [Charter](../../docs/GDI-SOP_charter.md#4-introduction).

### 7. Summary or Context Diagram

```mermaid
flowchart TD
    A[Approved dataset withdrawal request] --> B
    B{Confirm withdrawal request and identify dataset}
    B -->|Soft-deletion| C[Remove dataset record]
    B -->|Hard-deletion| D[Remove dataset and variants records]
    C --> E[Remove permissions for the dataset]

>     D --> E

Reviewer: In Scope and in 8.4 you mentioned data validation: _validating frequency integrity_ 

It's missing from this flowchart

Reviewer 2: Agreed

Author: Ah, validating frequency integrity didn't have to appear in the Scope section. I will remove it from there, thanks!

    E --> F[Log action]
    F --> G[Report completion to VHD]
```

### 8. Procedure

#### 8.1. Confirm withdrawal request and identify dataset

| Step identifier | When | Who |
| :-------------- | :------------------------------------------------------------------ | :-------------------------------------- |

> | `1`             | When a dataset withdrawal request has been approved under the overarching withdrawal process described in [GDI-SOP0009 dataset withdrawal](../european-level/GDI-SOP0009_dataset-withdrawal.md#87-per-system-dataset-withdrawal). | Node aggregated beacon maintainer |

Reviewer 2: ```suggestion
| `1`             | When a dataset withdrawal request has been approved under the overarching withdrawal process described in [GDI-SOP0009 dataset withdrawal](../european-level/GDI-SOP0009_dataset-withdrawal.md#87-per-system-dataset-withdrawal). | Node's AF Beacon maintainer |
```

As the node aggregated beacon maintainer, confirm that the incoming request package is complete before performing any deletion in beacon. Double check first:
- The original approved withdrawal request or ticket reference

> - The requested scope of the withdrawal (soft or hard-deletion)

Reviewer 2: This is not so much on the SOP0011, but more regarding the SOP0009 - Would you say we should always recommend soft-deletion by default to users requesting data withdrawal? I'm wondering if in your experience users are not aware of the differences, and when they come to the GDI VHD they may be confused and try to always go through the hard-deletion route, which is more cumbersome for GDI maintainers 🤔

Author: Well, hard-deletion frees up space and is cleaner than soft-deletion, IMO. I think that soft-deletion is meant for special cases where an owner just wants to remove the dataset temporary, maybe because some fix/update.

Reviewer 2: On the other hand, hard deletion goes against FAIR principles, unless a clear tombstone/placeholder record is made available under the dataset ID, which doesn't tend to happen

Author: True, FAIR principles are not respected here, but even for the soft-deletion. I expect to address it when we find an agreement for this PR notes item: There is no way in the Beacon specification that allows a beacon instance to currently show a metadata response with the history of datasets that existed and were removed. We need a clearer discussion before this can be retrieved via a simple beacon query.

- The relevant dataset identifier matching the node FAIR Data Point’s dataset to be withdrawn
Send an HTTP GET request to your beacon’s datasets endpoint and locate the record corresponding to the ID of the dataset to be deleted,
```bash

> curl 'https://<yourBeaconDomain>/datasets'

Reviewer 2: Wondering if ``curl`` is the default method, or we should add a disclaimer as in "use the method you prefer (e.g., curl, postman...)..."

Author: True, I just gave a solution, but there are many other possible solutions... I added a disclaimer as you suggested.

```
or perform the id query directly,
```bash
curl 'https://<yourBeaconDomain>/datasets/<id>'
```

> Record the dataset identifier and the deletion type (soft or hard). Record the dataset title only as an additional verification check.

Reviewer 2: What do you mean with "Record" here? is it to record it in the communications to the VHD? or is it as in "Write down because you'll need it later"?

Author: Yes, it's because you'll need it later for tha auditing.

Reviewer 2: ```suggestion
Take note of the dataset identifier and the deletion type (soft or hard), as you will need it later. Similarly, record the dataset title only as an additional verification check.
```

- If the request package is complete and the dataset is found, proceed to ⏩[Step 2](#82-remove-the-dataset-from-the-beacon-database).

> - If the request package is incomplete or the dataset is not found, request clarification from the GDI Virtual Helpdesk and pause the workflow until the missing information is provided.

Reviewer 2: ```suggestion
- If the request package is incomplete or the dataset is not found, request clarification from the GDI Virtual Helpdesk and pause the workflow until clarification is provided.
```
Since it may be that the dataset is not in this system, and thus there may be no missing information for the node maintainer, but rather that the withdrawal of the specified dataset doesn't need to go through this system

| Step identifier | When | Who |
| :-------------- | :------------------------------------------------------------------ | :-------------------------------------- |
| `2`             | After successful completion of [Step 1](#81-confirm-withdrawal-request-and-identify-dataset). | Node aggregated beacon maintainer |

After deletion type of the dataset withdrawal is confirmed:
- If the deletion type is a soft-deletion, proceed to ⏩[Step 2.1](#821-soft-deletion).
- If the deletion type is a hard-deletion, proceed to ⏩[Step 2.2](#822-hard-deletion-option-a).

##### 8.2.1. Soft-deletion

| Step identifier | When | Who |
| :-------------- | :--------------------------------------- | :-------------------------------------- |
| `2.1`             | The deletion type was identified as soft. | Node aggregated beacon maintainer |

Locate yourself in the server where your beacon is running and execute the following command line script to remove the dataset record approved for  withdrawal.
```bash
docker exec mongoprod /bin/bash -c 'mongosh beacon -u <user> -p <password> --authenticationDatabase admin --eval "db.datasets.deleteMany({\"id\": \"<id>\"})"'
```

> 

Reviewer: You have redinex in both hard deletions, but not here. Is this correct?

Author: Yes, because in the soft one, the variants are never removed 👍

Reviewer 2: If variants are never removed, then can the effects of the presence of that data propagate to other datasets?

In other words, the effective deletion of the dataset would not be real, as it would not change the AFs?

Author: The AFs are statically calculated, not on the fly. There's confusion about this, because we are not allowing progressive updates. Everything works as a single package. This is what was agreed in GDI, as for now. So, no, no propagation possible.

> - If the removal is successful, proceed to ⏩[Step 3](#83-remove-dataset-permissions).

Reviewer 2: We need to explicitly say how to check if removal is successful. This could be a paragraph here.

If the verification in the hard-deletion is similar to this, I would personally add then another step in between, where both soft/hard deletion redirects to and checks that the removal was successful.

#### 8.3. Remove dataset permissions

| Step identifier | When | Who |
| :-------------- | :--------------------------------------- | :-------------------------------------- |
| `3`             | After ⏩[Step 2](#82-remove-the-dataset-from-the-beacon-database). | Node aggregated beacon maintainer |

Located at the server where your beacon is running, proceed to edit the file at the path `/beacon/permissions/datasets/datasets_permissions.yml` and remove the dataset entry.
Save the file.
- Continue to ⏩[Step 4](#84-verify-log-and-report-completion).

#### 8.4. Verify, log, and report completion

| Step identifier | When | Who |
| :-------------- | :--------------------------------------- | :-------------------------------------- |
| `4`             | After successful completion of Step 3 | Node aggregated beacon maintainer |

Verify in the beacon that the withdrawn records in the dataset are not appearing in a query of datasets and a query of variants.
Add the dataset id to the file in path `/beacon/conf/datasets/datasets/datasets_conf.yml` and add a new item under it called isDeprecated setting it as True:

> ```yaml

Reviewer: maybe extend this entry with date, deletion type, who put in request?

Author: This should go to audit file, not in the conf file, so no need to add anything else.

When attempting a withdrawal of the beacon records related to a dataset, it will be necessary to determine which is the exact ID of the dataset, as case sensitivity and other character inferences may result in an unsuccessful withdrawal attempt. The withdrawal procedure of a soft-deletion is slightly different from a hard-deletion, where the amount of records to be deleted is significantly higher. Withdrawal by hard-deletion includes variants and dataset records type, whereas soft-deletion includes only the latter. 
It is important that the withdrawal procedure is preceded by a double-check of the dataset requested for withdrawal in coordination with the Fair Data Point service.
For a broader context of GDI SOPs, please refer to the [Charter](../../docs/GDI-SOP_charter.md#4-introduction).

> ### 7. Summary or Context Diagram

Reviewer 2: I took the liberty to create a more detailed diagram, to explain the logic of the rest of my comments, and trying to follow the format of the GDI SOP0009. Feel free to copy-paste it, or edit it as you see fit.

````mermaid
flowchart TB
 subgraph s1["<b>Node Allele Frequency Beacon Maintainer</b>"]
        n1["Confirm withdrawal<br>request and identify dataset"]
        n2["Package complete and<br>dataset found?"]
        n3["Request clarification<br>from VHD and pause"]
        n4["Deletion<br>type?"]
        n5["Soft-deletion:<br>Remove dataset record"]
        n6["Removal<br>successful?"]
        n7["Hard-deletion Option A:<br>use remove_dataset.py<br>+ reindex"]
        n8["Removal<br>successful?"]
        n9["Hard-deletion Option B:<br>Manual delete datasets<br>+ variants + reindex"]
        n11["Record failure details<br>and report to VHD"]
        n12["Remove dataset<br>permissions"]
        n13["Verify withdrawn data<br>not in beacon queries"]
        n14["Add isDeprecated<br>flag to config"]
        n15["Record in audit<br>/ change log"]
        n16["Report completion<br>to VHD"]
  end

    start("<i>Start</i>") L_start_n1@--> n1
    n1 L_n1_n2@--> n2
    n2 L_n2_n4@-- yes --> n4
    n2 -- no --> n3
    n3 -- "clarification<br>received" --> n1
    n4 L_n4_n5@-- soft --> n5
    n4 L_n4_n7@-- hard --> n7
    n5 L_n5_n6@--> n6
    n6 L_n6_n12@-- yes --> n12
    n6 -- no --> n11
    n7 L_n7_n8@--> n8
    n8 L_n8_n12@-- yes --> n12
    n8 L_n8_n9@-- no --> n9
    n9 L_n9_n10@--> n6
    n11 L_n11_n12@--> n12
    n12 L_n12_n13@--> n13
    n13 L_n13_n14@--> n14
    n14 L_n14_n15@--> n15
    n15 L_n15_n16@--> n16
    n16 L_n16_end@--> ending("<i><b>End</b></i>")

    n2@{ shape: decision }
    n4@{ shape: decision }
    n6@{ shape: decision }
    n8@{ shape: decision }

    style s1 fill:#BBDEFB
    style ending fill:#757575
    style start fill:#FFD600

    L_start_n1@{ animation: slow }
    L_n1_n2@{ animation: slow }
    L_n2_n4@{ animation: slow }
    L_n4_n5@{ animation: slow }
    L_n4_n7@{ animation: slow }
    L_n5_n6@{ animation: slow }
    L_n6_n12@{ animation: slow }
    L_n7_n8@{ animation: slow }
    L_n8_n12@{ animation: slow }
    L_n8_n9@{ animation: slow }
    L_n9_n10@{ animation: slow }
    L_n11_n12@{ animation: slow }
    L_n12_n13@{ animation: slow }
    L_n13_n14@{ animation: slow }
    L_n14_n15@{ animation: slow }
    L_n15_n16@{ animation: slow }
    L_n16_end@{ animation: slow }
````

Author: Added, thanks!

As the node aggregated beacon maintainer, confirm that the incoming request package is complete before performing any deletion in beacon. Double check first:
- The original approved withdrawal request or ticket reference
- The requested scope of the withdrawal (soft or hard-deletion)
- The relevant dataset identifier matching the node FAIR Data Point’s dataset to be withdrawn
Send an HTTPS GET request to your beacon’s datasets endpoint and locate the record corresponding to the ID of the dataset to be deleted,
```bash
curl 'https://<yourBeaconDomain>/datasets'
```
or perform the ID beacon query directly,
```bash
curl 'https://<yourBeaconDomain>/datasets/<id>'
```
Record the dataset identifier and the deletion type (soft or hard). Record the dataset title only as an additional verification check.
- If the request package is complete and the dataset is found, proceed to ⏩[Step 2](#82-remove-the-dataset-from-the-beacon-database).
- If the request package is incomplete or the dataset is not found, request clarification from the GDI Virtual Helpdesk and pause the workflow until the missing information is provided.

> #### 8.2. Remove the dataset from the beacon database

Reviewer 2: Along the process there are useful commands but I'm missing a direct link to the, very likely, existing documentation on the matter. Even if it's just "I'm not explaining what this does, but if you want to know, go here" kind of thing, that would be helpful

Reviewer 2: For example, I don't know the difference between Hard deletion option A and B; and if there was an error on deletion A, whether to go straight to option B or try to interpret it.

If I have the specs/docs at hand, it would be better.

Author: But I think I made it clear that you have to start by option A and then if not successful, go for option B.

Reviewer 2: Yes, I was referring to generic documentation links that could help the person going through it. Either as a reference at the bottom of the file, or as an embedded URL in here. To the spec that I think I referenced somewhere else in this PR

| `2`             | After successful completion of [Step 1](#81-confirm-withdrawal-request-and-identify-dataset). | Node aggregated beacon maintainer |

After deletion type of the dataset withdrawal is confirmed:
- If the deletion type is a soft-deletion, proceed to ⏩[Step 2.1](#821-soft-deletion).
- If the deletion type is a hard-deletion, proceed to ⏩[Step 2.2](#822-hard-deletion-option-a).

##### 8.2.1. Soft-deletion

| Step identifier | When | Who |
| :-------------- | :--------------------------------------- | :-------------------------------------- |
| `2.1`             | The deletion type was identified as soft. | Node aggregated beacon maintainer |

Locate yourself in the server where your beacon is running and execute the following command line script to remove the dataset record approved for  withdrawal.
```bash
docker exec mongoprod /bin/bash -c 'mongosh beacon -u <user> -p <password> --authenticationDatabase admin --eval "db.datasets.deleteMany({\"id\": \"<id>\"})"'
```

- If the removal is successful, proceed to ⏩[Step 3](#83-remove-dataset-permissions).

> - If the removal is not successful, record the response obtained from the used commands, adding all the information about the actions performed and the intended goal of performing them and report to the GDI Virtual Helpdesk so that requester communication continues through the VHD workflow and proceed to ⏩[Step 3](#83-remove-dataset-permissions).

Reviewer 2: In the case that it's not successful, would you expect then the node AF beacon maintainer to just report to the VHD and move to remove permissions? Or would they need to wait until clarification is given before going to step 3?

I'm asking because the phrase as of now implies parallel work

Author: I think they can go to remove dataset permissions straight away, which will at least vanish data for the dataset in case an API call is made. After clarification is given on the failure, then, they can finish removing data from the database. But yes, I think is a parallel work and makes sense, thanks for the good question!

##### 8.2.2. Hard-deletion (option A)

| Step identifier | When | Who |
| :-------------- | :--------------------------------------- | :-------------------------------------- |

> | `2.2`             | The deletion type was identified as hard. | Node aggregated beacon maintainer |

Reviewer 2: Similar to my comment above, change the "Who" in these to the Node AF Beacon maintainer.

Reviewer 2: Just to be consistent with the names

Reviewer 2: Unless you meant aggregated as in "Both the normal Beacon and the AF Beacon" 🤔 
But then it would need to be disclosed in the terminology or somewhere else that you're referring to that

Located at the server where your beacon is running, execute the following command line script on the terminal adding the ID for the dataset to be removed with the -d flag.
```bash

> docker exec ri-tools python remove_dataset.py -d <id>

Reviewer 2: Haven't seen this ri-tools container mentioned somewhere else. Will the maintainer know it by default? Is there only one, and there's no room for human error?

Author: It's mandatory to be deployed during the official "step by step beacon deployment" document. Another thing is that the users remove it afterwards. But I think the AF Beacon node maintainers should know that there's this container.

Located at the server where your beacon is running, execute the following command line scripts to remove the dataset record and variants records related to the dataset to withdraw.
```bash
docker exec mongoprod /bin/bash -c 'mongosh beacon -u <user> -p <password> --authenticationDatabase admin --eval "db.datasets.deleteMany({\"id\": \"<id>\"})"'
docker exec mongoprod /bin/bash -c 'mongosh beacon -u <user> -p <password> --authenticationDatabase admin --eval "db.genomicVariations.deleteMany({\"datasetId\": \"<id>\"})"'
```
Next, execute the following command line script on the terminal to remove index references for that dataset.
```bash
docker exec beaconprod python -m beacon.connections.mongo.reindex
```
- If the removal is successful, proceed to ⏩[Step 3](#83-remove-dataset-permissions).
- If the removal is not successful, record the response obtained from the used commands, adding all the information about the actions performed and the intended goal of performing them and report to the GDI Virtual Helpdesk so that requester communication continues through the VHD workflow and proceed to ⏩[Step 3](#83-remove-dataset-permissions).

> #### 8.3. Remove dataset permissions

Reviewer 2: After editing ``datasets_permissions.yml`` and ``datasets_conf.yml``, is a beacon service restart needed for changes to take effect?

Author: No, it's applied dynamically. No need to restart.

| Step identifier | When | Who |
| :-------------- | :--------------------------------------- | :-------------------------------------- |
| `3`             | After ⏩[Step 2](#82-remove-the-dataset-from-the-beacon-database). | Node aggregated beacon maintainer |

> Located at the server where your beacon is running, proceed to edit the file at the path `/beacon/permissions/datasets/datasets_permissions.yml` and remove the dataset entry.

Reviewer 2: I would add a bit more details on this. For example, what to look for; what to remove exactly (provide an example code block of what it looks like and what to remove).

Save the file.
- Continue to ⏩[Step 4](#84-verify-log-and-report-completion).

#### 8.4. Verify, log, and report completion

| Step identifier | When | Who |
| :-------------- | :--------------------------------------- | :-------------------------------------- |

> | `4`             | After successful completion of Step 3 | Node aggregated beacon maintainer |

Reviewer 2: The issue I have with the current steps, is that I think we're missing one that is for failure. Otherwise, steps 2 redirect you to step 3 on failure of withdrawal, and then you come here and it's phrased as if the withdrawal was successful, and to verify the records are not appearing. Hence why I would have a step of failure in the diagram/procedure, to be explicit with the logic

Author: I added this: After ⏩[Step 3.1](#831-soft-deletion) or  ⏩[Step 3.2](#832-hard-deletion), maybe with this is not necessary to add a new Step just for failure?

> Verify in the beacon that the withdrawn records in the dataset are not appearing in a beacon query of datasets and a beacon query of variants.

Reviewer 2: There needs to be more details on how to assess this.

Author: After adding a new step `8.3. Check that the dataset was removed successfully` I think this might not be needed anymore, so I removed this part: `Verify in the beacon that the withdrawn records in the dataset are not appearing in a beacon query of datasets and a beacon query of variants.`. I would add a new action item with checking that metadata with the removed dataset appears saying the dataset is deprecated, but this is still to see how we do it in beacon as I mentioned before, this is not in the spec and I have to figure out how to do it via a beacon query.

| Withdrawal    | An exceptional action where a dataset (or research object) is removed from public access, typically retaining its persistent identifier and replacing access with a notice (often via a tombstone page/record)                                        |

### 3. Roles and Responsibilities

See qualifications and responsibilities of the roles at the [**Organisational Roles and Responsibilities**](../../docs/GDI-SOP_organisational-roles-and-responsibilities.md) document.

| Role       | Full name                   | GDI/node role                                    | Organisation                          |
| ---------- | --------------------------- | ------------------------------------------------ | ------------------------------------- |
| Author     | Liina Nagirnaja             | Beacon Manager                                   | CRG                                   |
| Author     | Jordi Rambla                | Beacon Product Owner                             | CRG                                   |
| Author     | Oriol López-Doriga Sagalés  | Beacon Developer                                 | CRG                                   |

> | Author     | Aleš Čep                    | Task 4.3 member                                  | CRG                                   |

Reviewer: I am not sure why my name is added as Author. This must be mistake. If any, maybe change it to Reviewer. Also, my organization is not CRG, it is a University Of Maribor (UM)

Author: Apologies @AlesCUMSI , changed it now!

```bash
curl 'https://<yourBeaconDomain>/datasets/<id>'
```
- If the response was empty, the dataset was removed successfully, proceed to ⏩[Step 4](#84-remove-dataset-permissions). 
- If the response was not empty, the dataset removal was not successful, record the response obtained from the used commands, adding all the information about the actions performed and the intended goal of performing them and report to the GDI Virtual Helpdesk so that requester communication continues through the VHD workflow and proceed to ⏩[Step 4](#84-remove-dataset-permissions).

##### 8.3.2. Hard-deletion

| Step identifier | When | Who |
| :-------------- | :--------------------------------------- | :-------------------------------------- |
| `3.2`             | After the removal to perform was identified as a hard-deletion in ⏩[Step 3](#83-check-that-the-dataset-was-removed-successfully) | AF beacon maintainer |

Execute a HTTPS GET request to your beacon looking for the dataset. Use the method you prefer (e.g., curl, postman...)..."
```bash
curl 'https://<yourBeaconDomain>/datasets/<id>/g_variants'

> ```

Reviewer 3: Is this sufficient to check that both the dataset record and the variants have been removed successfully? Will not this query return an empty set also if only the variants has been removed from the dataset, but not the dataset entry itself? And for a hard deletion, both variants and dataset records should be removed.

Author: Absolutely, you are right. The hard-deletion can't be checked just using the API, so I have changed the command for something internal using mongoDB commands.

Reviewer 3: Thanks Oriol, but did you accidently add the beacon variants queries under the Soft-deletion verification section instead of the hard-deletion verification section?

Author: Oh yes, sorry @kjellp ! I have now addressed it if you want to double check.

Reviewer 3: Looks good!

| Author     | Liina Nagirnaja             | Beacon Manager                                   | CRG                                   |
| Author     | Jordi Rambla                | Beacon Product Owner                             | CRG                                   |
| Author     | Oriol López-Doriga Sagalés  | Beacon Developer                                 | CRG                                   |
| Reviewer     | Òscar Martínez Llobet  | Task 4.3 member                                  | CRG                                   |
| Reviewer     | Kjell Petersen  | Task 4.3 member                                  | University of Bergen                                   |

> | Reviewer     | Nejc Nadizar  | Task 4.3 member                                  | University of Ljubljana                                |

Reviewer 2: ````suggestion
| Approver | Òscar Martínez Llobet  | Task 4.3 member                                  | CRG                                   |
| Approver | Kjell Petersen  | Task 4.3 member                                  | University of Bergen                                   |
| Approver | Nejc Nadizar  | Task 4.3 member                                  | University of Ljubljana                                |
````
